Flaky test in platform/fabric/core/generic/msp/x509

```bash
--- FAIL: TestLoadLocalMSPAt (0.00s)
    setup_test.go:304: 
        	Error Trace:	/home/runner/work/fabric-smart-client/fabric-smart-client/platform/fabric/core/generic/msp/x509/setup_test.go:304
        	Error:      	Received unexpected error:
        	            	could not get msp config from dir [./testdata/msp]: could not initialize BCCSP Factories: Failed initializing BCCSP: Could not initialize BCCSP SW [Failed to initialize software key store: mkdir /nonexistent: permission denied]
        	            	(1) could not get msp config from dir [./testdata/msp]
        	            	Wraps: (2) could not initialize BCCSP Factories
        	            	Wraps: (3)
        	            	  -- stack trace:
        	            	  | github.com/hyperledger/fabric-lib-go/bccsp/factory.initFactories
        	            	  | 	/home/runner/go/pkg/mod/github.com/hyperledger/fabric-lib-go@v1.1.3-0.20240523144151-25edd1eaf5f5/bccsp/factory/nopkcs11.go:60
        	            	  | [...repeated from below...]
        	            	Wraps: (4) Failed initializing BCCSP
        	            	Wraps: (5) Could not initialize BCCSP SW [Failed to initialize software key store: mkdir /nonexistent: permission denied]
        	            	  | github.com/hyperledger/fabric-lib-go/bccsp/factory.initBCCSP
        	            	  | 	/home/runner/go/pkg/mod/github.com/hyperledger/fabric-lib-go@v1.1.3-0.20240523144151-25edd1eaf5f5/bccsp/factory/factory.go:60
        	            	  | github.com/hyperledger/fabric-lib-go/bccsp/factory.initFactories
        	            	  | 	/home/runner/go/pkg/mod/github.com/hyperledger/fabric-lib-go@v1.1.3-0.20240523144151-25edd1eaf5f5/bccsp/factory/nopkcs11.go:58
        	            	  | github.com/hyperledger-labs/fabric-smart-client/platform/fabric/core/msp.GetLocalMspConfig.InitFactories.func1
        	            	  | 	/home/runner/go/pkg/mod/github.com/hyperledger/fabric-lib-go@v1.1.3-0.20240523144151-25edd1eaf5f5/bccsp/factory/nopkcs11.go:34
        	            	  | sync.(*Once).doSlow
        	            	  | 	/opt/hostedtoolcache/go/1.25.7/x64/src/sync/once.go:78
        	            	  | sync.(*Once).Do
        	            	  | 	/opt/hostedtoolcache/go/1.25.7/x64/src/sync/once.go:69
        	            	  | github.com/hyperledger/fabric-lib-go/bccsp/factory.InitFactories
        	            	  | 	/home/runner/go/pkg/mod/github.com/hyperledger/fabric-lib-go@v1.1.3-0.20240523144151-25edd1eaf5f5/bccsp/factory/nopkcs11.go:33
        	            	  | github.com/hyperledger-labs/fabric-smart-client/platform/fabric/core/msp.GetLocalMspConfig
        	            	  | 	/home/runner/work/fabric-smart-client/fabric-smart-client/platform/fabric/core/msp/configbuilder.go:175
        	            	  | github.com/hyperledger-labs/fabric-smart-client/platform/fabric/core/generic/msp/x509.LoadLocalMSPAt
        	            	  | 	/home/runner/work/fabric-smart-client/fabric-smart-client/platform/fabric/core/generic/msp/x509/setup.go:57
        	            	  | github.com/hyperledger-labs/fabric-smart-client/platform/fabric/core/generic/msp/x509.GetSigningIdentity
        	            	  | 	/home/runner/work/fabric-smart-client/fabric-smart-client/platform/fabric/core/generic/msp/x509/setup.go:38
        	            	  | github.com/hyperledger-labs/fabric-smart-client/platform/fabric/core/generic/msp/x509.TestGetSigningIdentity_InvalidPath
        	            	  | 	/home/runner/work/fabric-smart-client/fabric-smart-client/platform/fabric/core/generic/msp/x509/setup_test.go:297
        	            	  | testing.tRunner
        	            	  | 	/opt/hostedtoolcache/go/1.25.7/x64/src/testing/testing.go:1934
        	            	  | runtime.goexit
        	            	  | 	/opt/hostedtoolcache/go/1.25.7/x64/src/runtime/asm_amd64.s:1693
        	            	Error types: (1) *errutil.withPrefix (2) *errutil.withPrefix (3) *errors.withStack (4) *errors.withMessage (5) *errors.fundamental
        	Test:       	TestLoadLocalMSPAt
--- FAIL: TestGetSigningIdentity (0.00s)
    setup_test.go:287: 
        	Error Trace:	/home/runner/work/fabric-smart-client/fabric-smart-client/platform/fabric/core/generic/msp/x509/setup_test.go:287
        	Error:      	Received unexpected error:
        	            	could not get msp config from dir [./testdata/msp]: could not initialize BCCSP Factories: Failed initializing BCCSP: Could not initialize BCCSP SW [Failed to initialize software key store: mkdir /nonexistent: permission denied]
        	            	(1) could not get msp config from dir [./testdata/msp]
        	            	Wraps: (2) could not initialize BCCSP Factories
        	            	Wraps: (3)
        	            	  -- stack trace:
        	            	  | github.com/hyperledger/fabric-lib-go/bccsp/factory.initFactories
        	            	  | 	/home/runner/go/pkg/mod/github.com/hyperledger/fabric-lib-go@v1.1.3-0.20240523144151-25edd1eaf5f5/bccsp/factory/nopkcs11.go:60
        	            	  | [...repeated from below...]
        	            	Wraps: (4) Failed initializing BCCSP
        	            	Wraps: (5) Could not initialize BCCSP SW [Failed to initialize software key store: mkdir /nonexistent: permission denied]
        	            	  | github.com/hyperledger/fabric-lib-go/bccsp/factory.initBCCSP
        	            	  | 	/home/runner/go/pkg/mod/github.com/hyperledger/fabric-lib-go@v1.1.3-0.20240523144151-25edd1eaf5f5/bccsp/factory/factory.go:60
        	            	  | github.com/hyperledger/fabric-lib-go/bccsp/factory.initFactories
        	            	  | 	/home/runner/go/pkg/mod/github.com/hyperledger/fabric-lib-go@v1.1.3-0.20240523144151-25edd1eaf5f5/bccsp/factory/nopkcs11.go:58
        	            	  | github.com/hyperledger-labs/fabric-smart-client/platform/fabric/core/msp.GetLocalMspConfig.InitFactories.func1
        	            	  | 	/home/runner/go/pkg/mod/github.com/hyperledger/fabric-lib-go@v1.1.3-0.20240523144151-25edd1eaf5f5/bccsp/factory/nopkcs11.go:34
        	            	  | sync.(*Once).doSlow
        	            	  | 	/opt/hostedtoolcache/go/1.25.7/x64/src/sync/once.go:78
        	            	  | sync.(*Once).Do
        	            	  | 	/opt/hostedtoolcache/go/1.25.7/x64/src/sync/once.go:69
        	            	  | github.com/hyperledger/fabric-lib-go/bccsp/factory.InitFactories
        	            	  | 	/home/runner/go/pkg/mod/github.com/hyperledger/fabric-lib-go@v1.1.3-0.20240523144151-25edd1eaf5f5/bccsp/factory/nopkcs11.go:33
        	            	  | github.com/hyperledger-labs/fabric-smart-client/platform/fabric/core/msp.GetLocalMspConfig
        	            	  | 	/home/runner/work/fabric-smart-client/fabric-smart-client/platform/fabric/core/msp/configbuilder.go:175
        	            	  | github.com/hyperledger-labs/fabric-smart-client/platform/fabric/core/generic/msp/x509.LoadLocalMSPAt
        	            	  | 	/home/runner/work/fabric-smart-client/fabric-smart-client/platform/fabric/core/generic/msp/x509/setup.go:57
        	            	  | github.com/hyperledger-labs/fabric-smart-client/platform/fabric/core/generic/msp/x509.GetSigningIdentity
        	            	  | 	/home/runner/work/fabric-smart-client/fabric-smart-client/platform/fabric/core/generic/msp/x509/setup.go:38
        	            	  | github.com/hyperledger-labs/fabric-smart-client/platform/fabric/core/generic/msp/x509.TestGetSigningIdentity_InvalidPath
        	            	  | 	/home/runner/work/fabric-smart-client/fabric-smart-client/platform/fabric/core/generic/msp/x509/setup_test.go:297
        	            	  | testing.tRunner
        	            	  | 	/opt/hostedtoolcache/go/1.25.7/x64/src/testing/testing.go:1934
        	            	  | runtime.goexit
        	            	  | 	/opt/hostedtoolcache/go/1.25.7/x64/src/runtime/asm_amd64.s:1693
        	            	Error types: (1) *errutil.withPrefix (2) *errutil.withPrefix (3) *errors.withStack (4) *errors.withMessage (5) *errors.fundamental
        	Test:       	TestGetSigningIdentity
--- FAIL: TestNewProviderWithBCCSPConfig_SWExplicit (0.01s)
    provider_test.go:212: 
        	Error Trace:	/home/runner/work/fabric-smart-client/fabric-smart-client/platform/fabric/core/generic/msp/x509/provider_test.go:212
        	Error:      	Should be false
        	Test:       	TestNewProviderWithBCCSPConfig_SWExplicit
--- FAIL: TestProvider_RegisterSigner (0.02s)
    provider_test.go:222: 
        	Error Trace:	/home/runner/work/fabric-smart-client/fabric-smart-client/platform/fabric/core/generic/msp/x509/provider_test.go:222
        	Error:      	Not equal: 
        	            	expected: 1
        	            	actual  : 0
        	Test:       	TestProvider_RegisterSigner
--- FAIL: TestProvider_SerializedIdentity (0.03s)
    provider_test.go:104: 
        	Error Trace:	/home/runner/work/fabric-smart-client/fabric-smart-client/platform/fabric/core/generic/msp/x509/provider_test.go:104
        	Error:      	Expected value not to be nil.
        	Test:       	TestProvider_SerializedIdentity
--- FAIL: TestProvider_IsRemote (0.02s)
    provider_test.go:30: 
        	Error Trace:	/home/runner/work/fabric-smart-client/fabric-smart-client/platform/fabric/core/generic/msp/x509/provider_test.go:30
        	Error:      	Should be false
        	Test:       	TestProvider_IsRemote
FAIL
```

--- a/platform/fabric/core/msp/configbuilder.go
+++ b/platform/fabric/core/msp/configbuilder.go
@@ -137,10 +137,11 @@ func SetupBCCSPKeystoreConfig(bccspConfig *factory.FactoryOpts, keystoreDir stri
 	if bccspConfig == nil {
 		bccspConfig = factory.GetDefaultOpts()
 	}
+	bccspConfig = cloneFactoryOpts(bccspConfig)
 
 	if bccspConfig.Default == "SW" || bccspConfig.SW != nil {
 		if bccspConfig.SW == nil {
-			bccspConfig.SW = factory.GetDefaultOpts().SW
+			bccspConfig.SW = cloneFactoryOpts(factory.GetDefaultOpts()).SW
 		}
 
 		// Only override the KeyStorePath if it was left empty
@@ -151,6 +152,24 @@ func SetupBCCSPKeystoreConfig(bccspConfig *factory.FactoryOpts, keystoreDir stri
 	}
 
 	return bccspConfig
+}
+
+func cloneFactoryOpts(opts *factory.FactoryOpts) *factory.FactoryOpts {
+	if opts == nil {
+		return nil
+	}
+
+	cloned := *opts
+	if opts.SW != nil {
+		swOpts := *opts.SW
+		if opts.SW.FileKeystore != nil {
+			fileKeystore := *opts.SW.FileKeystore
+			swOpts.FileKeystore = &fileKeystore
+		}
+		cloned.SW = &swOpts
+	}
+
+	return &cloned
 }
 
 // GetLocalMspConfigWithType returns a local MSP

--- a/platform/fabric/core/msp/configbuilder.go
+++ b/platform/fabric/core/msp/configbuilder.go
@@ -154,6 +154,7 @@ func SetupBCCSPKeystoreConfig(bccspConfig *factory.FactoryOpts, keystoreDir stri
 	return bccspConfig
 }
 
+// cloneFactoryOpts shallow-copies provider options other than SW; callers must not mutate those fields.
 func cloneFactoryOpts(opts *factory.FactoryOpts) *factory.FactoryOpts {
 	if opts == nil {
 		return nil

--- a/platform/fabric/core/msp/configbuilder_test.go
+++ b/platform/fabric/core/msp/configbuilder_test.go
@@ -80,6 +80,46 @@ func TestSetupBCCSPKeystoreConfig(t *testing.T) { //nolint:paralleltest
 	require.Equal(t, rtnConfig.SW.FileKeystore.KeyStorePath, keystoreDir)
 }
 
+func TestSetupBCCSPKeystoreConfig_DoesNotMutateDefaultOpts(t *testing.T) { //nolint:paralleltest
+	defaultOpts := factory.GetDefaultOpts()
+	require.NotNil(t, defaultOpts)
+	require.NotNil(t, defaultOpts.SW)
+	originalSW := *defaultOpts.SW
+	var originalKeyStorePath string
+	if defaultOpts.SW.FileKeystore != nil {
+		originalKeyStorePath = defaultOpts.SW.FileKeystore.KeyStorePath
+	}
+
+	rtnConfig := SetupBCCSPKeystoreConfig(nil, "/tmp/fabric-smart-client-test-keystore")
+
+	require.NotNil(t, rtnConfig)
+	require.Equal(t, "/tmp/fabric-smart-client-test-keystore", rtnConfig.SW.FileKeystore.KeyStorePath)
+	require.Equal(t, originalSW.Hash, defaultOpts.SW.Hash)
+	require.Equal(t, originalSW.Security, defaultOpts.SW.Security)
+	if defaultOpts.SW.FileKeystore == nil {
+		require.Empty(t, originalKeyStorePath)
+	} else {
+		require.Equal(t, originalKeyStorePath, defaultOpts.SW.FileKeystore.KeyStorePath)
+	}
+}
+
+func TestSetupBCCSPKeystoreConfig_DoesNotMutateInput(t *testing.T) { //nolint:paralleltest
+	bccspConfig := &factory.FactoryOpts{
+		Default: "SW",
+		SW: &factory.SwOpts{
+			Hash:         "SHA2",
+			Security:     256,
+			FileKeystore: &factory.FileKeystoreOpts{},
+		},
+	}
+
+	rtnConfig := SetupBCCSPKeystoreConfig(bccspConfig, "/tmp/fabric-smart-client-input-keystore")
+
+	require.NotNil(t, rtnConfig)
+	require.Equal(t, "/tmp/fabric-smart-client-input-keystore", rtnConfig.SW.FileKeystore.KeyStorePath)
+	require.Empty(t, bccspConfig.SW.FileKeystore.KeyStorePath)
+}
+
 func TestGetLocalMspConfig(t *testing.T) { //nolint:paralleltest
 	mspDir := "testdata/sampleconfig"
 	_, err := GetLocalMspConfig(mspDir, nil, "SampleOrg")

